src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -789,5 +789,66 @@ public bool IsWixBinary
                 return this.isWixBinary.Value;
             }
         }
+
+        public bool IsChecksumAlgorithmSecureForPortablePdb()
+        {
+            const string sha256 = "8829d00f-11b8-4213-878b-770e8597ac16";
+            var sha256guid = new Guid(sha256);
+
+            if (this.peReader.TryOpenAssociatedPortablePdb(
+                this.FileName,
+                filePath => File.Exists(filePath) ? File.OpenRead(filePath) : null,
+                out MetadataReaderProvider pdbProvider,
+                out _))
+            {
+                MetadataReader metadataReader = pdbProvider.GetMetadataReader();
+                foreach (DocumentHandle document in metadataReader.Documents)
+                {
+                    Document doc = metadataReader.GetDocument(document);
+
+                    Guid hashGuid = metadataReader.GetGuid(doc.HashAlgorithm);
+
+                    return hashGuid == sha256guid;
+                }
+            }
+
+            return false;

```

Comment:
>return false [](start = 12, length = 12)

Always prefer early exit if possible. So, change the code to say if (!TryOpen) then return false immediately.